### DISCUSSION TITLE
Lattice2 and Part Design workbench

### DISCUSSION THREAD

**Discussion by macdroid53** [DeepSOIC/Lattice2#60] [Q&A]
Trying to figure out how to use the Lattice Part Design Pattern.
Do I need to place an array of paper planes in the Body?

No matter what I select in the Body before attempting an array, it says the axis link must be an edge.
If I select an edge of the Body, the array is off in space somewhere.

Probably obvious, but, I'm missing something.

**Comment by macdroid53** [comment]
Got it to work...not sure if this is the right way though.

I just made sure nothing was selected before clicking to create an array.

**Comment by DeepSOIC** [comment]
Yes, it is necessary to put the array into the body. For that to happen, the body must be activated (but not selected), it is indicated by bold font and permanent highlight in model tree. To activate/deactivate a body, double-click it in model tree.

It is not possible to drag an array into a body if the array was created outside. It's a design quirk, it is because Body was not designed to have these foreign things inside. Lattice2 arrays that are made when body is active pretend to be sketches to sneak into bodies. Normal arrays don't pretend to be sketches.

**Reply by macdroid53** [reply]
> Yes, it is necessary to put the array into the body. For that to happen, the body must be activated (but not selected), it is indicated by bold font and permanent highlight in model tree. To activate/deactivate a body, double-click it in model tree.


That typical Part Design operation.

> 
> It is not possible to drag an array into a body if the array was created outside. It's a design quirk, it is because Body was not designed to have these foreign things inside. Lattice2 arrays that are made when body is active pretend to be sketches to sneak into bodies. Normal arrays don't pretend to be sketches.

So, this something like  PDWrapper so it can live in the Body container. Got it.

Thanks!
